addons/mail/models/mail_mail.py

```diff
@@ -197,10 +208,19 @@ def _postprocess_sent_message(self, success_pids, failure_reason=False, failure_
                     # TDE TODO: could be great to notify message-based, not notifications-based, to lessen number of notifs
                     messages._notify_message_notification_update()  # notify user that we have a failure
         if not failure_type or failure_type in ['mail_email_invalid', 'mail_email_missing']:  # if we have another error, we want to keep the mail.
-            mail_to_delete_ids = [mail.id for mail in self if mail.auto_delete]
-            self.browse(mail_to_delete_ids).sudo().unlink()
+            mails_to_delete = self.filtered(lambda mail: mail.auto_delete)
+            if mails_to_delete:
+                mails_to_delete.to_delete = True
+
         return True
 
+    @api.autovacuum
+    def _gc_mail_mail(self):
+        self.env['mail.mail'].search([
+            ('auto_delete', '=', True),
```

Comment:
Would remove this to speed up the request even more.

I don't see when we would have records flagged as to_delete that don't have auto_delete.
Besides, it would be strange letting those just sit in the database anyway.